charts/taraxa-node/templates/initconfig-node.yaml

```diff
@@ -0,0 +1,84 @@
+    VRF="TARAXA_NODE_VRF_${INDEX}"
+
+    KEY="${!KEY}"
+    VRF="${!VRF}"
+
+    if [ -z "$KEY" ]
+    then
+      if [ ! -f "$WALLET_PATH" ]
+      then
+        echo "No predifined keys. Generating new wallet..."
+        KEY=$(taraxad --command account | grep node_secret | cut -d\  -f3- | tr -d \")
+        VRF=$(taraxad --command vrf | grep vrf_secret | cut -d\  -f3 | tr -d \")
+      else
+        echo "Found wallet file."
+        KEY=$(cat "$WALLET_PATH" | jq -r .node_secret)
+        VRF=$(cat "$WALLET_PATH" | jq -r .vrf_secret)
+      fi
+    fi
+
+    echo "Generating config"
+    taraxad --command config \
+      --network-id {{ .Values.config.network }} \
+      --node-secret ${KEY} \
+      --vrf-secret ${VRF} \
+      {{- if and (ne .Values.config.network "1") (ne .Values.config.network "2") (ne .Values.config.network "3") }}
+      {{- $lbEnabled := .Values.bootnode.loadBalancer.enabled -}}
+      {{- $lbIPs := .Values.bootnode.loadBalancer.addresses -}}
+      {{- range $key, $value := .Values.config.bootnode.addresses }}
+      --boot-nodes {{ if $lbEnabled }}{{ index $lbIPs $key }}{{- else }}{{ include "taraxa-boot-node.fullname" $ }}-{{ $key }}.{{ include "taraxa-boot-node.fullname" $ }}.{{$.Release.Namespace}}{{- end }}:10002/{{ $value }} \
+      {{- end }}
+      {{- end }}
+      {{- if .Values.config.extraArgs }}
+      {{ join " " .Values.config.extraArgs }} \
+      {{- end }}
+      --config $CONFIG_PATH \
+      --wallet $WALLET_PATH \
+      --data-dir $DATA_PATH
+
+    {{- if and (ne .Values.config.network "1") (ne .Values.config.network "2") (ne .Values.config.network "3") }}
+    export DELEGATION_ADDRESS=$(cat $CONFIG_PATH | jq -r '.chain_config.final_chain.state.dpos.genesis_state | keys[0]')
+    export FIRST_NODE_ADDRESS=$(cat $CONFIG_PATH | jq -r '.chain_config.final_chain.state.dpos.genesis_state[env.DELEGATION_ADDRESS] | keys[0]')
+    export DELEGATION_VALUE=$(cat $CONFIG_PATH | jq -r '.chain_config.final_chain.state.dpos.genesis_state[env.DELEGATION_ADDRESS][env.FIRST_NODE_ADDRESS]')
+    cat $CONFIG_PATH | jq '.chain_config.final_chain.state.dpos.genesis_state[env.DELEGATION_ADDRESS] = ({})' > $CONFIG_PATH.tmp && mv $CONFIG_PATH.tmp $CONFIG_PATH
+    {{- range $key, $value := .Values.config.consensusnode.keys }}
+    echo "Adding node {{ $key }} address to the config..."
+    export ADDRESS="0x$(taraxad --command account {{ $value }} | grep "node_address" | awk '{print $3}' | tr -d '"')"
+    cat $CONFIG_PATH | jq '.chain_config.final_chain.state.dpos.genesis_state[env.DELEGATION_ADDRESS] += ({(env.ADDRESS): env.DELEGATION_VALUE})' > $CONFIG_PATH.tmp && mv $CONFIG_PATH.tmp $CONFIG_PATH
+    {{- end }}
+    {{- end }}
+
+    {{- if and (ne .Values.config.network "1") (ne .Values.config.network "2") (ne .Values.config.network "3") }}
+    {{ if .Values.explorer.enabled }}
+    export FAUCET_ADDRESS=$(taraxad --command account {EXPLORER_FAUCET_PRIVATE_KEY} | grep node_address | cut -d\  -f3 | tr -d \")
```

Comment:
If i understood correctly, this should be
`${EXPLORER_FAUCET_PRIVATE_KEY}`

please double check variable expansion

Reply:
This is the same like with previous value.

```
    export FAUCET_ADDRESS=$(taraxad --command account {{ .Values.explorer.faucet.privKey }} | grep node_address | cut -d\  -f3 | tr -d \")

```
It was like that. I move that to `Secret`

Reply:
same typo

Reply:
haha, thanks! Havent seen that. It worked for me, because I was providing Secret as SealeSecret. Will fix tomorrow.

Reply:
fixed